rainforest/resources.go

```diff
@@ -12,50 +12,67 @@ type Folder struct {
 	Title string `json:"title"`
 }
 
-// GetFolders returns a slice of Folders (their names and IDs) which are available
-// for filtering RF tests.
-func (c *Client) GetFolders() ([]Folder, error) {
-	// Prepare request
-	req, err := c.NewRequest("GET", "folders?page_size=100", nil)
+// getPaginatedResource gets all of a resource from endpoint (using multiple
+// requests, if necessary) and returns the result.
+//
+// Because of the type system, usage is hairy. coll should be a pointer to an
+// empty slice of the appropriate type, and collect is called every time
+// resources are added to the collection. The caller should handle collecting
+// the collection.
+func (c *Client) getPaginatedResource(endpoint string, coll interface{}, collect func(interface{})) error {
```

Comment:
Instead of using a callback, can't we have `getPaginatedResource` return a blank interface and just handle the return value in the caller? Not really sure if that's any less gross though...

Reply:
I tried soooo many things, couldn't get anything to work other than this :sob:. The problem is `interface{}` and `[]interface{}` work differently and it's extremely hard to move things in between.